### PR TITLE
Development

### PR DESCRIPTION
**Pull request by jpavlav**
Take a look, let me know.

### CODE REVIEW

File: alertmanager.py [hunk 1]
```diff
@@ -28,45 +28,71 @@ def request_session(self):
...
-
-    def get_silence(self):
+        route = "/api/v1/groups"
+        r = self._make_request("GET", route)
+        if self._check_response(r):
+            return Alert.from_dict(r.json())
+
+    def get_silence(self, id=None):
+        route = "/api/v1/silences"
+        if id:
```
Comment (Kamori) [inline alertmanager.py]: https://docs.python.org/2/library/functions.html#id

We should probably avoid built ins for variable naming

File: alertmanager.py [hunk 2]
```diff
@@ -28,45 +28,71 @@ def request_session(self):
...
 
-    def post_silence(self):
-        # http://10.255.238.146:9093/api/v1/silences
-        pass
-
-    def delete_silence(self):
-        # http://10.255.238.146:9093/api/v1/silence/:id:
-        pass
+    def delete_silence(self, id):
+        route = "/api/v1/silence/{0}".format(id)
```
[the remark this answers is not in the document]
Comment (jpavlav) [reply alertmanager.py]: Yes, I'll take care of it.

File: alertmanager.py [hunk 3]
```diff
@@ -28,45 +28,71 @@ def request_session(self):
 
         return self._req_obj
 
+    # Just made this so that I could quickly finish this before i leave
+    def _check_response(response):
+        """Helper to check that responses are what we expect"""
+        if 'success' in r.json():
+            return True
+        else:
+            raise Exception('You did it wrong')
```
[the remark this answers is not in the document]
Comment (jpavlav) [reply alertmanager.py]: And yes, ValueError would probably be good here. This again, was done really quickly, so I hadn't put much though into what specific Exception we should employ. I'll update.

File: alertmanager.py [hunk 4]
```diff
@@ -28,45 +28,71 @@ def request_session(self):
 
         return self._req_obj
 
+    # Just made this so that I could quickly finish this before i leave
+    def _check_response(response):
```
[the remark this answers is not in the document]
Comment (jpavlav) [reply alertmanager.py]: Good, I like it too :).

File: alertmanager.py (hunk 1 above)
Comment (jpavlav) [reply alertmanager.py]: You're right, a mistake made in haste.

File: alertmanager.py [hunk 5]
```diff
@@ -28,45 +28,71 @@ def request_session(self):
...
+        """Helper to check that responses are what we expect"""
+        if 'success' in r.json():
```
[the remark this answers is not in the document]
Comment (jpavlav) [reply alertmanager.py]: This was just a quick stand-in. I think that it's likely we should be checking both the response data itself, as well as the response code associated with the request. That way, we can make sure, yes it went through, and yes, we got back what we wanted. If we don't get back what we wanted, we can raise an appropriate Exception an maybe include the json response in the Exception as well.

**Comment by Kamori** [comment]
:+1: LGTM... the other stuff we can clear up as we go!

keep up the good work

**Pull request merged**